freqtrade/exchange/exchange.py

```diff
@@ -2283,6 +2303,146 @@ async def _fetch_funding_rate_history(
+                                      columns=combined_df.columns)
+                # Age out old candles
+                if first_required_candle_date:
+                    # slice of older dates
+                    trades_df = trades_df[
+                        first_required_candle_date < trades_df['timestamp']]
+                    trades_df = trades_df.reset_index(drop=True)
+            self._trades[(pair, timeframe, c_type)] = trades_df
+        return trades_df
+
+    def refresh_latest_trades(self,
+                              pair_list: ListPairsWithTimeframes,
+                              data_handler: Any,  # using IDataHandler ends with circular import
+                              *,
+                              cache: bool = True,
+                              ) -> Dict[PairWithTimeframe, DataFrame]:
+        """
+        Refresh in-memory TRADES asynchronously and set `_trades` with the result
+        Loops asynchronously over pair_list and downloads all pairs async (semi-parallel).
+        Only used in the dataprovider.refresh() method.
+        :param pair_list: List of 3 element tuples containing (pair, timeframe, candle_type)
+        :param since_ms: time since when to download, in milliseconds
+        :param cache: Assign result to _trades. Usefull for one-off downloads like for pairlists
+        :param drop_incomplete: Control candle dropping.
+            Specifying None defaults to _ohlcv_partial_candle
+        :return: Dict of [{(pair, timeframe): Dataframe}]
+        """
+        logger.debug("Refreshing TRADES data for %d pairs", len(pair_list))
+        since_ms = None
+        results_df = {}
+        for pair, timeframe, candle_type in set(pair_list):
+            new_ticks: List = []
+            all_stored_ticks_df = DataFrame(columns=DEFAULT_TRADES_COLUMNS + ['date'])
+            first_candle_ms = self.needed_candle_ms(timeframe, candle_type)
+            # refresh, if
+            # a. not in _trades
+            # b. no cache used
+            # c. need new data
+            is_in_cache = (pair, timeframe, candle_type) in self._trades
+            if (not is_in_cache
+                    or not cache
+                    or self._now_is_time_to_refresh_trades(pair, timeframe, candle_type)):
+                logger.debug(f"Refreshing TRADES data for {pair}")
+                # fetch trades since latest _trades and
+                # store together with existing trades
+                try:
+                    until = None
+                    from_id = None
+                    if is_in_cache:
+                        from_id = self._trades[(pair, timeframe, candle_type)].iloc[-1]['id']
+                        until = dt_ts()  # now
+
+                    else:
+                        until = int(timeframe_to_prev_date(timeframe).timestamp()) * 1000
+                        all_stored_ticks_df = data_handler.trades_load(f"{pair}-cached")
+
```

Comment:
What's the reason to use a "-cached" filename here?

Reply:
I was trying to avoid filename collisions. You think we don't need it?

Reply:
well i don't know - what's the reason to have this "duplicated"? 

like - is there a problem we expect by appending to the already existing files? 
Sure, it could delay initial startup (avoid holes in the data) ... 

But on the other hand, assume someone would like to backtest over a period of a year - and has a dry-run running at the same time ...
at the end, they'd have the same exact data twice - once in -cached files, once without.